jupyterlab_celltests/lint.py

```diff
@@ -68,33 +65,30 @@ def run(notebook, executable=None, rules=None):
     rules = rules or {}
     extra_metadata.update(rules)
 
+    # TODO: lintfail is more like lintpassed?
+
     if 'lines_per_cell' in extra_metadata:
-        lines_per_cell = extra_metadata.get('lines_per_cell', -1)
-        lintret, lintfail = lint_lines_per_cell(lines_per_cell, extra_metadata)
+        lintret, lintfail = lint_lines_per_cell(extra_metadata['cell_lines'], max_lines_per_cell=extra_metadata['lines_per_cell'])
```

Comment:
Repeated default values/overly defensive style was a bit confusing to me (and repeated defaults are at risk of conflicting/drifting apart):

* `'cell_lines'` is guaranteed to exist in `extra_metadata` because of the earlier call to `extract_extrametadata()`. So no need for the get with default (`metadata.get('cell_lines', [])`) in `lint_lines_per_cell()` (above).

* There's already a check that `'lines_per_cell'` is in `extra_metadata`, so no need for the get with default (`extra_metadata.get('lines_per_cell', -1)`).

Also applies to each of the similar sections below.